clifford/jit_func.py

```diff
@@ -0,0 +1,92 @@
+
+import ast
+try:
+    import astpretty
+    AST_PRETTY_AVAILABLE = True
+except ImportError:
+    AST_PRETTY_AVAILABLE = False
+import inspect
+import warnings
+import textwrap
+from ._numba_utils import njit
+from ._ast_transformer import DecoratorRemover, GATransformer
+
+
+class jit_func(object):
+    """
+    This is a JIT decorator that re-writes the AST and then numba JITs
+    the resulting function.
+    """
+    def __init__(self, layout, ast_debug=False, mv_constants={}, scalar_constants={}):
+        self.layout = layout
+        self.mv_constants = mv_constants
+        self.scalar_constants = scalar_constants
+        if AST_PRETTY_AVAILABLE:
+            self.ast_debug = ast_debug
+        else:
+            if ast_debug:
+                warnings.warn(
+                    'The ast_debug flag is set to True, but the astpretty module is not importable.\n'
+                    'To see ast_debug output please pip install astpretty')
+            self.ast_debug = False
+
+    def __call__(self, func):
+        # Get the function source
+        fname = func.__name__
+        source = inspect.getsource(func)
+        # Remove the indentation
+        source = textwrap.dedent(source)
+
+        # Parse the source
+        tree = ast.parse(source)
+        if self.ast_debug:
+            print('\n\n\n\n TRANSFORMING FROM \n\n\n\n')
+            astpretty.pprint(tree)
+
+        # Remove the decorators from the function
+        tree = DecoratorRemover().visit(tree)
```

Comment:
Strictly you want to leave behind all the non-`jitfunc` decorators, but that's fine as a todo.